### PR TITLE
get_provision_type - move from miq-RedHat-Satellite. Determine the provision type to use for the miq_provision in the VMProvision_*/* state machine.

### PR DESCRIPTION
**Pull request by itewk**

### CODE REVIEW

File: Automate/RedHatConsulting_Utilities/Infrastructure/VM/Lifecycle.class/__methods__/get_provision_type.rb [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Determine the provision type to use for the miq_provision in the VMProvision_*/* state machine
```
Comment (curt-matthews) [inline Automate/RedHatConsulting_Utilities/Infrastructure/VM/Lifecycle.class/__methods__/get_provision_type.rb]: /for the miq_provision/for miq_provision

File: Automate/RedHatConsulting_Utilities/Infrastructure/VM/Lifecycle.class/__methods__/get_provision_type.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#
+# The default behavior of the "out-of-the-box" state machine of using $evm.root['miq_provision'].provision_type
```
Comment (curt-matthews) [inline Automate/RedHatConsulting_Utilities/Infrastructure/VM/Lifecycle.class/__methods__/get_provision_type.rb]: /machine of using/machine is using/

File: Automate/RedHatConsulting_Utilities/Infrastructure/VM/Lifecycle.class/__methods__/get_provision_type.rb [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+# can be overridden if $evm.root['miq_provision'].get_option(:custom_provision_type) is set.
+#
+# EXPECTED
+#   EVM ROOT
+#     miq_provision - Provisining request to determine the provision type for
```
Comment (curt-matthews) [inline Automate/RedHatConsulting_Utilities/Infrastructure/VM/Lifecycle.class/__methods__/get_provision_type.rb]: /Provisining/Provisioning/

File: Automate/RedHatConsulting_Utilities/Infrastructure/VM/Lifecycle.class/__methods__/get_provision_type.rb (hunk 1 above)
Comment (itewk) [reply Automate/RedHatConsulting_Utilities/Infrastructure/VM/Lifecycle.class/__methods__/get_provision_type.rb]: why removing the `the`, i am refering to a specific miq_provision in this case, the current one.

File: Automate/RedHatConsulting_Utilities/Infrastructure/VM/Lifecycle.class/__methods__/get_provision_type.rb (hunk 3 above)
Comment (itewk) [reply Automate/RedHatConsulting_Utilities/Infrastructure/VM/Lifecycle.class/__methods__/get_provision_type.rb]: fixed in latest PR

File: Automate/RedHatConsulting_Utilities/Infrastructure/VM/Lifecycle.class/__methods__/get_provision_type.rb (hunk 1 above)
Comment (itewk) [reply Automate/RedHatConsulting_Utilities/Infrastructure/VM/Lifecycle.class/__methods__/get_provision_type.rb]: fixed in latest PR

**Pull request merged**